src/coreComponents/common/LogLevelsInfo.hpp

```diff
@@ -0,0 +1,171 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file LogLevelsInfo.hpp
+ */
+#ifndef GEOS_COMMON_LOGLEVELSINFO_HPP
+#define GEOS_COMMON_LOGLEVELSINFO_HPP
+
+#include "DataTypes.hpp"
+#include "format/Format.hpp"
+
+namespace geos
+{
+
+namespace logInfo
+{
+
+/**
+ * @brief Trait used to check whether a LOG_LEVEL_INFO structure is valid.
+ * @tparam LOG_LEVEL_INFO The log level structure to check.
+ */
+template< typename LOG_LEVEL_INFO >
+static constexpr bool is_log_level_info =
+  std::is_same_v< int, decltype(LOG_LEVEL_INFO::getMinLogLevel()) > &&
+  std::is_same_v< std::string_view, decltype(LOG_LEVEL_INFO::getDescription()) >;
+
+/** ThOSE 3 macros would replace the ones in Logger.hpp  */
+/**
+ * @brief Output messages based on current Group's log level.
```

Comment:
I would add a region to wrap the ways to use the LogLevelsInfo (macros, potencially `isLogLevelActive()`).